### PR TITLE
Christy's promisify simple-db (lab2)

### PR DESCRIPTION
**Pull request by christylaguardia**

### CODE REVIEW

File: lib/store.js [hunk 1]
```diff
@@ -1,70 +1,57 @@
...
         const filename = path.join(this.root, object._id + '.json');
+        const contents = JSON.stringify(object);
 
-        fs.writeFile(filename, JSON.stringify(object), err => {
-            if (err) return callback(err);
-            callback(null, object);
-        });
+        return writeFile(filename, contents)
+            .then(() => object)
+            // .catch(err => console.log(err));
```
Comment (nsmeds) [inline lib/store.js]: Did you comment out this `catch` for a reason? It's good practice to always include a `catch` (though not necessary in Mocha tests, as Marty pointed out).

File: tests/db.test.js [hunk 2]
```diff
@@ -1,169 +1,133 @@
...
-                done();
-            });
+        it('new book with JSON content', () => {
+            let newBook = { author: 'Dr. Seuss', title: 'The Cat in the Hat' };
+            return books.save(newBook)
+                .then(book => {
+                    assert.equal(book.author, newBook.author);
+                    assert.equal(book.title, newBook.title);
+                    assert.ok(book._id);
+                    // QUESTION: add readFile assert?
```
Comment (nsmeds) [inline tests/db.test.js]: This test looks fine as is; I don't think you need to bring in fs.readFile.

File: tests/db.test.js [hunk 3]
```diff
@@ -1,169 +1,133 @@
...
-                // assert.deepEqual(files, [web, cat, charlie]); // doesn't work
-                assert.equal(files.length, 3);
-                done();
-            });
+        it('books and returns array', () => {
+            let book1 = { author: 'Roald Dahl', title: 'Charlie and the Chocolate Factory' };
+            let book2 = { author: 'C. S. Lewis', title: 'The Lion, the Witch and the Wardrobe' };
+            
+            //QUESTION: correct way to chain?
+            return books.save(book1)
```
Comment (nsmeds) [inline tests/db.test.js]: This chaining looks good to me.

File: tests/db.test.js [hunk 4]
```diff
@@ -1,169 +1,133 @@
...
+                .then(books.save(book2))
+                .then(books.getAll(files => {
+                    //QUESTION: how can I test the array?
+                    assert.equal(files.length, 2);
```
Comment (nsmeds) [inline tests/db.test.js]: Marty suggested using `sort()` here to sort the array items by id. Testing `files.length` is ok for an end-to-end test, although it wouldn't hurt to get more in-depth.

File: lib/store.js (hunk 1 above)
Comment (christylaguardia) [inline lib/store.js]: I think @martypdx said I should not use it here?

Comment (nsmeds) [inline lib/store.js]: You're right, not necessary here - nevermind that comment.

**Pull request merged**